### PR TITLE
Handle NS and NSE server wait timeouts cleanly

### PR DESCRIPTION
**Pull request by acnodal-tc**
gRPC server-side streaming calls should be closed by returning nil so we need to figure out what the timeout Error looks like and return nil when it happens. In this case the Error has an "unknown" code so we use the error message. When it happens we return nil and the server-side streaming call is closed cleanly.

Signed-off-by: Toby Cabot <toby@acnodal.io>

### CODE REVIEW

**Review by denis-tingaikin — COMMENTED** [review]
Added a few cosmetic comments. In general, looks good

File: pkg/registry/etcd/ns_server.go [hunk 1]
```diff
@@ -162,7 +162,13 @@ func (n *etcdNSRegistryServer) Find(query *registry.NetworkServiceQuery, s regis
 		}
 	}
 	if query.Watch {
-		if err := n.watch(query, s); err != nil && !errors.Is(err, io.EOF) {
+		if err := n.watch(query, s); err != nil {
+			// If we have timed out then return nil to close the stream
+			// cleanly
+			if errors.Is(err, io.EOF) || err.Error() == "context canceled" {
```
[the remark this answers is not in the document]
Comment (acnodal-tc) [reply pkg/registry/etcd/ns_server.go]: Yup!

**Comment by acnodal-tc** [comment]
I found a cleaner way to do this (in the watch()) function).

**Review by glazychev-art — APPROVED** [review]

**Review by denis-tingaikin — APPROVED** [review]
@acnodal-tc Many thanks!

**Review by denis-tingaikin — APPROVED** [review]

**Pull request merged**